core/src/main/java/com/google/cloud/sql/config/ConnectionConfig.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.sql.config;
```

Comment:
Do we need another package here? Could this live with our other public classes in com.google.cloud.sql?

Reply:
I want a separate package. This value object is going to become part of the public java api for #1266. I want to keep all new public java api classes and interfaces in `com.google.cloud.sql` or `com.google.cloud.sql.config`. Legacy public API and internal implementation will continue to live in  `com.google.cloud.sql.core`. This will also enable us to possibly use Java Modules in the future to make a clear, enforceable boundary between our public and internal classes.

This warrants further discussion of the details.

Reply:
To summarize elsewhere, we recognize that com.google.cloud.sql already has a public class. So it seems best to keep using that location for the public interface.

Reply:
Done.